src/OpenIddict.Core/Infrastructure/OpenIddictProvider.Authentication.cs

```diff
@@ -378,27 +396,23 @@ public partial class OpenIddictProvider<TUser, TApplication, TAuthorization, TSc
             var services = context.HttpContext.RequestServices.GetRequiredService<OpenIddictServices<TUser, TApplication, TAuthorization, TScope, TToken>>();
 
             // Remove the authorization request from the distributed cache.
-            if (!string.IsNullOrEmpty(context.Request.GetRequestId())) {
+            if (!string.IsNullOrEmpty(context.Request.RequestId)) {
                 // Note: the ApplyAuthorizationResponse event is called for both successful
                 // and errored authorization responses but discrimination is not necessary here,
                 // as the authorization request must be removed from the distributed cache in both cases.
-                await services.Options.Cache.RemoveAsync(OpenIddictConstants.Environment.Request + context.Request.GetRequestId());
+                await services.Options.Cache.RemoveAsync(OpenIddictConstants.Environment.Request + context.Request.RequestId);
             }
 
             if (!context.Options.ApplicationCanDisplayErrors && !string.IsNullOrEmpty(context.Response.Error) &&
                                                                  string.IsNullOrEmpty(context.Response.RedirectUri)) {
                 // Determine if the status code pages middleware has been enabled for this request.
-                // If it was not registered or disabled, let the OpenID Connect server middleware render
+                // If it was not registered or enabled, let the OpenID Connect server middleware render
                 // a default error page instead of delegating the rendering to the status code middleware.
                 var feature = context.HttpContext.Features.Get<IStatusCodePagesFeature>();
                 if (feature != null && feature.Enabled) {
                     // Replace the default status code to return a 400 response.
                     context.HttpContext.Response.StatusCode = 400;
 
-                    // Store the OpenID Connect response in the HTTP context to allow retrieving it
-                    // from user code (e.g from an ASP.NET Core MVC controller or a Nancy module).
-                    context.HttpContext.SetOpenIdConnectResponse(context.Response);
```

Comment:
FYI, this is something ASOS beta7 now does automatically before invoking the `ApplyAuthorizationResponse` event.